src/RamCondition.h

```diff
@@ -212,34 +276,94 @@ class RamNotExists : public RamCondition {
         }
         return true;
     }
+
+    /** Obtain list of child nodes */
+    std::vector<const RamNode*> getChildNodes() const override {
+        std::vector<const RamNode*> res;
+        for (const auto& cur : values) {
+            res.push_back(cur.get());
+        }
+        return res;
+    }
+
+    /** Create clone */
+    RamNotExists* clone() const override {
+        RamNotExists* res = new RamNotExists(relation);
+        for (auto& cur : values) {
+            RamValue* val = nullptr;
+            if (cur != nullptr) {
+                val = cur->clone();
+            }
+            res->values.push_back(std::unique_ptr<RamValue>(val));
+        }
+        return res;
+    }
+
+    /** Apply */
+    void apply(const RamNodeMapper& map) override {
+        for (auto& val : values) {
+            if (val != nullptr) {
+                val = map(std::move(val));
+            }
+        }
+    }
+
+protected:
+    /** Check equality */
+    bool equal(const RamNode& node) const override {
+        assert(dynamic_cast<const RamNotExists*>(&node));
+        const RamNotExists& other = static_cast<const RamNotExists&>(node);
+        return getRelation() == other.getRelation() && equal_targets(values, other.values);
+    }
 };
 
-/** check whether a given relation is empty or not*/
+/**
+ * Emptiness check for a relation
+ */
+// TODO: Rename to RamEmptyCheck
```

Comment:
TODOs should be in the form` // TODO (#issueid): text` or `// TODO (userid): text`

Reply:
Done.